packages/streamr-logger/include/streamr-logger/StreamrLogFormatter.hpp

```diff
@@ -0,0 +1,175 @@
+#ifndef STREAMER_LOGGER_STREAMRLOGFORMATTER_H
+#define STREAMER_LOGGER_STREAMRLOGFORMATTER_H
+
+#include <iostream>
+#include <folly/Format.h>
+#include <folly/logging/LogFormatter.h>
+#include <folly/logging/LogLevel.h>
+
+namespace {
+
+static constexpr int maxFileNameAndLineNumberLength{36};
+static constexpr folly::StringPiece fileNameAndLineNumberSeparator{": "};
+static constexpr auto separatorLength{
+    std::ssize(fileNameAndLineNumberSeparator)};
+static const std::string nonTruncatedFormatter =
+    "{}{}{} [{:04d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}.{}] ({:<" +
+    std::to_string(maxFileNameAndLineNumberLength) + "}): {}{}{}\n";
+static constexpr std::string_view truncatedFormatter =
+    "{}{}{} [{:04d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}.{}] ({: <*}{}{}): {}{}{}\n";
+
+struct LogLevelData {
+    folly::StringPiece logLevelName;
+    folly::StringPiece color;
+};
+
+// FATAL cannot be used in folly because it aborts. So CRITICAL is converted
+// to Streamr fatal
+constexpr LogLevelData getLogLevelData(const folly::LogLevel level) {
+    switch (level) {
+        case folly::LogLevel::DBG:
+            return {"TRACE", "\033[90m"};
+            break;
+        case folly::LogLevel::DBG0:
+            return {"DEBUG", "\033[34m"};
+            break;
+        case folly::LogLevel::INFO:
+            return {"INFO", "\033[32m"};
+            break;
+        case folly::LogLevel::WARN:
+            return {"WARN", "\033[33m"};
+            break;
+        case folly::LogLevel::ERR:
+            return {"ERROR", "\033[31m"};
+            break;
+        case folly::LogLevel::CRITICAL:
+            return {"FATAL", "\033[1;41m"};
+            break;
+        default:
+            return {"TRACE", "\033[90m"};
+            break;
+    }
+}
+
+} // namespace
+
+namespace streamr::logger {
+
+/*
+Format based on this Streamr log. The Filename and line number is fixed size
```

Comment:
Also take the additional fields into account in the format. (context info and the extra parameter of logger.info(), warn etc )